src/components/text-editor/prosemirror-adapter/plugins/table-plugin.ts

```diff
@@ -0,0 +1,42 @@
+import { tableNodes, tableEditing } from 'prosemirror-tables';
+import { Plugin } from 'prosemirror-state';
+
+export const getTableEditingPlugins = (tablesEnabled: boolean): Plugin[] => {
+    if (tablesEnabled) {
+        return [tableEditing()];
+    }
+
+    return [];
+};
+
+export const getTableNodes = () => {
+    return tableNodes({
+        tableGroup: 'block',
+        cellContent: 'block+',
+        cellAttributes: {
+            background: {
+                default: null,
+                getFromDOM: (dom) => {
+                    return dom.style.backgroundColor || null;
+                },
+                setDOMAttr: (value: string, attrs: Record<string, string>) => {
+                    if (value) {
+                        attrs.style =
+                            (attrs.style || '') + `background-color: ${value};`;
+                    }
+                },
+            },
+            color: {
+                default: null,
+                getFromDOM: (dom) => {
+                    return dom.style.color || null;
+                },
+                setDOMAttr: (value: string, attrs: Record<string, string>) => {
+                    if (value) {
+                        attrs.style = (attrs.style || '') + `color: ${value};`;
```

Comment:
Same here, we need to consider injection attacks.

Reply:
Just adding this for clarity in case anyone is reading in future. We have updated prosemirror-model to version 1.22.1 as per this [thread here](https://discuss.prosemirror.net/t/heads-up-xss-risk-in-domserializer/6572)